### PR TITLE
REP-5401 Environment Variable Substitution in Configuration Files

### PR DESCRIPTION
**Pull request by dmnjohns**

### CODE REVIEW

**Comment by dmnjohns** [comment]
Down to a single test failure after that last commit. It looks like there is a classpath collision for `org.objectweb.asm.tree.ClassNode`. JTwig has a dependency on Parboiled which has an ASM dependency. GlassFish also has an ASM dependency. The versions clearly do not match, and so we see something like:
> Caused by: java.lang.NoSuchMethodError: org.objectweb.asm.tree.ClassNode.<init>(I)V

File: repose-aggregator/src/docs/asciidoc/architecture/configuration.adoc [hunk 1]
```diff
@@ -25,3 +25,78 @@ The configuration files are checked for updates every 15 seconds.
...
+
+[IMPORTANT]
+====
+When substituting an environment variable, *Repose* strictly requires that the named environment variable exists.
+If not, *Repose* will throw an exception when processing the containing configuration file.
+====
+
+[IMPORTANT]
+====
+Updating an environment variable will *not* cause Repose to reload any configuration which references that environment variable.
```
Comment (adrianjgeorge) [inline repose-aggregator/src/docs/asciidoc/architecture/configuration.adoc]: something we should potentially look at fixing in the future.

Comment (dmnjohns) [reply repose-aggregator/src/docs/asciidoc/architecture/configuration.adoc]: Sure, I think we can do that. It should just be a matter of polling environment variables and checking if the map has changed.

**Comment by adrianjgeorge** [comment]
Missed a license somewhere

**Comment by dmnjohns** [comment]
Bleh. Sure enough. It's the `glassfish-web.xml` file, which is a XML test resource, which is probably why I didn't catch it. Now I am wondering why that did not seem to be required when this file was in the `artifacts:web-application` project. Especially since those files actually make it into artifacts, while this file does not (where it lives now).

**Comment by adrianjgeorge** [comment]
i think we have an exclusion setup in the other module

**Comment by adrianjgeorge** [comment]
it appears you broke some other stuff

**Comment by dmnjohns** [comment]
retest this please

**Comment by dmnjohns** [comment]
Great. So one flaky test with fixed, the other two failed once and are now passing, and now a fourth has popped up.

Not cool, tests. Not cool.

**Comment by dmnjohns** [comment]
retest this please

**Comment by dmnjohns** [comment]
retest this please

File: repose-aggregator/src/docs/asciidoc/architecture/configuration.adoc (hunk 1 above)
Comment (adrianjgeorge) [reply repose-aggregator/src/docs/asciidoc/architecture/configuration.adoc]: Well this turns out to not be a thing

Comment (dmnjohns) [reply repose-aggregator/src/docs/asciidoc/architecture/configuration.adoc]: Whoops! But we still want to leave that warning in place, right?

**Pull request merged**